### ISSUE TITLE
Do single and multi channel models for earthquake transformer use same weights?

### ISSUE THREAD

**Issue by firstkingofrome** [seisbench/seisbench#225]
For both earthquake transformer in_channels is an adjustable parameter which seisbench either sets to 1 or 3 based on the number of components available for the target seismic trace.

Does anyone know if the weight set used for the 1 and 3 channel versions of these models is the same or if it would matter if they arent?

For example in: https://github.com/seisbench/seisbench/blob/f15e422ba0bc2452ab9c9f45394756c1b6f00fbc/seisbench/models/eqtransformer.py#L193

line 193:
```
        self.conv_d = nn.Conv1d(
            in_channels=self.filters[0], out_channels=1, kernel_size=11, padding=5
        )

#which is later used in the encoder on line 392
        for in_channels, out_channels, kernel_size in zip(
            [input_channels] + filters[:-1], filters, kernel_sizes
        ):
            convs.append(
                nn.Conv1d(
                    in_channels, out_channels, kernel_size, padding=kernel_size // 2
                )
            )
```
This seems to be using the Conv1d layer to ensure that inputs are always mapped to one channel (this is also communicated by figure 1 in the eqt paper https://www.nature.com/articles/s41467-020-17591-w).

It appears to me that when this model was trained it would have needed to be trained on both 3 channel and 1 channel data, yielding a separate weight set for a 1 and 3 channel model, otherwise I dont see how the dimensionality matches up. 

But when I look at seisbench I just see an option to load up weights from one of the per-trained versions of earthquake transformer:
``
 sbm.EQTransformer.from_pretrained("original") 
``

Does seisbench just pad out other channels (when there is only one channel) or do some other trick to avoid having to keep 1 and 3 channel weights for these models or am I just fundamentally miss understanding how seisbench works?

Im trying to figure this out because I am picking a bunch of historical single component data and I want to ensure that I am using the most appropriate EQT weight set.  

Thank you, I really appreciate any in sites on this issue!

**Comment by yetinam** [comment]
Hi @firstkingofrome ,

Thanks for your question and the comprehensive description. Indeed, the pretrained eqtransformer weights in SeisBench are all using three input channels (except the OBS model). As far as I know, the same hold for the models published with the original eqtransformer. However, you can safely apply the models to single channel data. The data is then padded with zeros on the other channels (assuming you use the annotate or classify function). To make this work, the training employes a so-called channel dropout, i.e., in some examples randomly some channels are ignored. This makes the model robust against missing channels. Clearly, in any case where three channels are available, all three should be used.

Does that answer your question?

**Comment by firstkingofrome** [comment]
That does, thank you for the explanation!
